tonic/src/transport/service/discover.rs

```diff
@@ -12,12 +12,12 @@ use tokio::{stream::Stream, sync::mpsc::Receiver};
 
 use tower::discover::{Change, Discover};
 
+type BoxFuture<T, E> = Pin<Box<dyn Future<Output = Result<T, E>> + Send + 'static>>;
```

Comment:
we could move these to some `common.rs` file? I think hyper does this

Reply:
yup

Reply:
Done. I defined `BoxFuture` inline in `transport` for now since it's the only one I could find we can extract. It is also not used anywhere else other than codegen, but I guess that one can stay independent.